src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
+    if (openingComment) {
+      DOMChildrenOperations.removeDelimitedText(
```

Comment:
Seems to work fine. :) But not sure if `Array.isArray` is a very robust way to check for this case.